### ISSUE TITLE
`export foo {.docgen.}`: allows a re-exported symbol to be docgen'd in re-exporting module

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/RFCs#409]
## proposal
```nim
# in std/private/foo.nim
proc foo1*()=
  ## comment

# in std/other.nim
proc foo2*()=
  ## comment

# in std/bar.nim
import std/private/foo, std/other
export foo1 {.docgen.}
export foo2
```

* `nim doc std/bar` should show docs for `foo1` in module `bar` as indicated by `{.docgen.}`
* `nim doc std/bar` should not show docs for `foo2` in module `bar` because `{.docgen.}` is absent there; the pragma leaves up to client code whether to docgen again the symbol; in this case, it allows distinguishing whether the symbol "belongs" logically in `bar` (where std/private/foo.nim is an implementation detail), vs where it's merely re-exported in the 2nd case, where std/other is the logical owner of foo2.
* in either case, the symbol is also docgen'd in the original module, so that foo1 gets docgen'd twice (in foo and in bar)
* runnableExamples should only run in original module, and ignore {.docgen.}

/cc @a-mr

**Comment by konsumlamm** [comment]
Would `{.docgen.}` also work for reexporting modules?

**Comment by timotheecour** [comment]
> Would {.docgen.} also work for reexporting modules?

good point, yes, i clarified this; since {.docgen.} is explicit, it makes sense to allow this and is useful when a whole module's symbols need to be docgen'd

**Comment by a-mr** [comment]
I cannot imagine cases when original module is half-owner and half-not-owner. So we have 2 cases:

1. The owner is original module. All re-exported symbols are already in `Exports` section with links to their original module. It's a good, DRY way in such a case.
2. When an original module is not owner (just an implementation helper) then `{.docgen.}` does make a perfect sense.

So per-symbol granularity does not make sense IMHO, while per-module one is OK.

**Comment by timotheecour** [comment]
> So per-symbol granularity does not make sense IMHO, while per-module one is OK.

a helper module could define APIs for more than 1 module though
```nim
helper:
proc fn1*()=discard
proc fn2*()=discard
highlevel:
import helper; export fn1
lowlevel:
import helper; export fn2
```
also, the "owner" semantics is mostly convenience for users groking docs, the point is to provide a way to show docs for a re-exported symbol in a module so that a project can be reorganized transparently; note that as far as symbol ownership is concerned (PSym and owner), docgen would have no effect.

**Comment by a-mr** [comment]
> a helper module could define APIs for more than 1 module though

if it's a helper then it's still logical to attach the pragma to whole module, i.e. to import statement in this case:
```nim
import helper {.docgen.}; export fn1
```
